File: MyAlbum.WebSPA/Models/Album.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace MyAlbum.Models
{
    public class Album
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ICollection<Photo> Photos { get; set; }

        public User Author { get; set; }

        public Album()
        {
            Photos = new Collection<Photo>();
        }
    }
}

File: src/WebSPA.React.Identity/ClientApp/src/models/album.ts
import { Photo } from './photo';
import { User } from "./user";

export interface AlbumQuery {
    categoryId: number;
    hasLocation: boolean;
    authorUserName: string;
}

export class Album {
    id: number = 0;
    name: string | null = null;
    createdDate: Date | null = null;
    modifiedDate: Date | null = null;
    author: User | null = null;
    mainPhoto: Photo | null = null;
    subPhotos: (Photo | null)[] = [];
    photos: Photo[] = [];
}

export class SaveAlbum {
    id?: number;
    name?: string;
}